src/server.js

```diff
@@ -75,25 +81,13 @@ module.exports = function() {
 		}
 	});
 
-	manager.sockets = sockets;
-
-	const protocol = config.https.enable ? "https" : "http";
-	const host = config.host || "*";
-
-	log.info(`The Lounge ${colors.green(Helper.getVersion())} is now running \
-using node ${colors.green(process.versions.node)} on ${colors.green(process.platform)} (${process.arch})`);
-	log.info(`Configuration file: ${colors.green(Helper.CONFIG_PATH)}`);
-	log.info(`Available on ${colors.green(protocol + "://" + host + ":" + config.port + "/")} \
-in ${config.public ? "public" : "private"} mode`);
-	log.info(`Press Ctrl-C to stop\n`);
-
-	if (!config.public) {
-		if ("autoload" in config) {
-			log.warn(`Autoloading users is now always enabled. Please remove the ${colors.yellow("autoload")} option from your configuration file.`);
-		}
+	manager = new ClientManager();
 
-		manager.autoloadUsers();
-	}
+	new Promise((resolve) => {
+		identHandler = new Identification(resolve);
+	}).then(() => {
+		manager.init(identHandler, sockets);
+	});
```

Comment:
This seems weird to me, as you're manipulating 2 promises without using the resulting one.

It seems equivalent to:

```js
identHandler = new Identification(() => manager.init(identHandler, sockets));
```

A nicer way to do this IMO would be to have the constructor construct his thing only (see [this](http://stackoverflow.com/a/24686979/1935861)) then have another method that returns a promise.

Then you can do:

```js
identHandler = new Identification().start(); // Or whatever name is appropriate here
identHandler.then(() => manager.init(identHandler, sockets));
```

Not a fan of having a constructor do anything else than creating its own instance...